### PR TITLE
feat(compiler): support recovery of malformed property reads

### PR DESCRIPTION
**Pull request by ayazhafiz**
This feature is trivial to support since
89c5255b8ca59eed27ede9e1fad69857ab0c6f4f has landed.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Feature

### CODE REVIEW

File: packages/compiler/src/expression_parser/parser.ts [hunk 1]
```diff
@@ -825,7 +825,13 @@ export class _ParseAST {
   parseAccessMemberOrMethodCall(receiver: AST, isSafe: boolean = false): AST {
     const start = receiver.span.start;
     const nameStart = this.inputIndex;
-    const id = this.expectIdentifierOrKeyword();
+    const id = this.withContext(ParseContextFlags.Writable, () => {
+      const id = this.expectIdentifierOrKeyword();
+      if (id.length === 0) {
+        this.error(`Property access cannot be empty`, receiver.span.end);
```
Comment (alxhub) [inline packages/compiler/src/expression_parser/parser.ts:831]:
```suggestion
        this.error(`Expected identifier for property access`, receiver.span.end);
```

**Review by alxhub — APPROVED** [review]

**Comment by ayazhafiz** [comment]
Caretaker: could you run g3sync?

**Comment by AndrewKushnir** [comment]
@ayazhafiz sure, I've started a new [presubmit](http://test/OCL:338122855:BASE:338122883:1603224933774:2a974f6f).

Note: since this is a feature, I've also changed the target to "minor", so the change will be merged into master only and will be available in v11.1.0. Please let me know if this sounds correct and feel free to update the target as needed.

**Comment by ayazhafiz** [comment]
That sounds good, thank you for the update/

**Pull request closed without merging**